restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/proxy/ProxyManager.java

```diff
@@ -103,13 +104,18 @@ private Address contact(final Gateway gateway, final int expires) throws Servlet
         if (outboundInterface == null)
             outboundInterface = (SipURI) factory.createSipURI(null, address);
         final String user = gateway.getUserName();
+        String contactUser = new String (user);
+        if (contactUser.contains("@")) {
+            int index = contactUser.indexOf("@");
+            contactUser = contactUser.substring(0, index);
+        }
         final String host = outboundInterface.getHost();
         int port = outboundInterface.getPort();
         if (port == -1) {
             port = outboundInterface().getPort();
         }
         final StringBuilder buffer = new StringBuilder();
-        buffer.append("sip:").append(user).append("@").append(host).append(":").append(port);
+        buffer.append("sip:").append(contactUser).append("@").append(host).append(":").append(port);
```

Comment:
would be great to create meaningful constants fro "@" and ":" is all over the place...

Reply:
There is a lot of place where is using @ and : .
We need to create a common place to define all of these thing. It's would be great in another story for refactoring code.

Reply:
agree